juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule

```diff
@@ -0,0 +1,489 @@
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT033 {
+                event-id EACHIP_EVENT033;
+                filter "fpc%{NUMBER:fpc_number}.*EA.*gumem.*par_protect";
+                description "The \"gumem[.].par_protect\" message is caused by a transient hardware issue which is automatically repaired. Global microcode space is parity protected. This is caused by a transient hardware issue and is repaired by the software automatically.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If few occurrences, no action.  If numerous occurrences, then contact JTAC.";
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT034 {
+                event-id EACHIP_EVENT034;
+                filter "fpc%{NUMBER:fpc_number}.*EA.*ucode sb data error";
+                description "The \"ucode sb data error\" message is caused by a transient hardware issue, which is automatically repaired. This error is caused by a transient hardware defect. Local microcode sb(single-bit) or db(double-bit) memory errors are repaired from the shadow memory space.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If few occurrences, no action.  If numerous occurrences, then contact JTAC.";
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT035 {
+                event-id EACHIP_EVENT035;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*FO.*Parity error detected for output buffer control";
+                description "The \"Parity error detected for output buffer control\" message reports a transient hardware error in EACHIP Fabric output block. This is due to a transient hardware issue within the EACHIP Fabric output block.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "If only a few errors occur, then clear the Alarm.  Else, contact JTAC.";
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT036 {
+                event-id EACHIP_EVENT036;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*FI.*Parity error detected for bank.*of CP table";
+                description "The \"Parity error detected for bank .. of CP table\" message reports a hardware parity error detected in the EACHIP Fabric input block. This is due to a transient hardware SRAM error within EACHIP Fabric input block.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```

Comment:
Should we also have a count based rule for this?

Reply:
@ConstrainedRandomVar , added trigger based on count